### ISSUE TITLE
Creation of Expense Manager function for the project

### ISSUE THREAD

**Issue by steelwolf180** [steelwolf180/Midas#8]
Hi there, I am thinking of adding the expanses manager function for the user to fill in their expenses on a daily basis according to the categories that had is shown below. After that we could consider integration of storing it in a database (MySQL/Sqllite).

1) Savings/Investments
2) Spending
3) Charity
4) Learning
5) Expenses
6) Loans

**Comment by Chuphay** [comment]
While I think having these pre-organized expenses is very simple for the
end user to use, they also perhaps don't have as much flexibility as real
life demands.
What I think may be a better idea, is to let the end user decide what
categories he/she wants to focus on. For example loans means nothing to me,
but food and transportation mean a lot to me. And I'd rather have the
software reflect what I think is important, rather than it being dictated
to me, if that makes sense.

**Comment by steelwolf180** [comment]
Chuphay, I totally agree with what you mean for a expenses manager function it needs to have the flexibility of each person to customize their own since everyone's spending habits is different.  

What i am proposing is that we follow what is calculated from our own salary and then set it as a sort of budget for each categories and then add expenses according to those categories. Maybe my further explanation of the reason for each categories would help you understand so that we could be on the same page. 

1) Savings/Investments - For your emergency fund and investment to grow your money (Never to be touched except for really emergency)
2) Spending - To use for enjoyment so to allow you to have the motivation to earn more and to prevent yourself from touching your savings/investment.
3) Charity - To give back to the society for what you have gained from it.
4) Learning - To allow the person to upgrade their skills, knowledge and mindset to be competitive in the market
5) Expenses - To pay for daily expenses like food, transport, utilities, bills
6) Loans - For repayment of your debt.

We could change the application to allow the user to start small and then convert it to the full blown categories once they are confident. The small steps may like these below.  

1) Savings/Investments - 10%
2) Spending - 10%
2) Expenses - 80%

**Comment by Chuphay** [comment]
two thoughts on this.
First, how about if when you first start the program, you get a prompt that
asks about your monthly income, where you get a couple of choices like:
salary, fixed income investments, retirement funds, etc, but also allows
you to input other sources of income (and also the prompt tells you that
you can change this later if you want). Then the prompt asks about your
expenses and again you have some fixed choices, but you can also fill in
your own blanks.
I'm not sure how easy it would be to program this and it's probably a lot
easier just to start with the simplest base case, just so that we can just
get started, but we should keep in mind what we will want the program to do
in the future.
second, you got me thinking with looking at are own salaries as an example.
I work part-time, and as such, every month my income varies wildly, and
while I don't live paycheck to paycheck, a couple times a year, I do have
to think about the liquid money I have available (i.e. during the summer I
basically have zero income and have to plan accordingly in the spring)

**Comment by steelwolf180** [comment]
Chuphay, regarding on your first thought it's really similar to the expenses manager that i am currently using in my phone. But further I modify it by adding additional accounts to align to the financial plan that i set out for myself.

https://play.google.com/store/apps/details?id=com.expensemanager&hl=en

Regarding on your 2nd thought, my proposal to start from small to distribute only about 10% to your savings/investment and spending, then for the remaining 80% of their income to their personal expenses. I believe it works for people who are both working part-time with varie income and those who works full time with stable incomes. 

I can't confirm that it works for everyone but it definitely works for me to allow me to accumulate 3 mths worth of my income and growing it through the small step.

I still think it will be much easier to program a basic financial plan of say 3 types for the user to choose from. Than giving the user multiple choices for simplicity sake and ease of use

**Comment by Chuphay** [comment]
I mean, cool man... get it going. that dude, Jonathan Humphreys, has already put in a lot of work. I already supplied some nice code and some good suggestions. Let's see some product! I mean, almost anything... but dont expect the two of us to do all the work for you. and by the way, simply making Jon's code into an .exe is not work. that only takes a few seconds.

we're behind you, but you need to take the lead on this project, and to be honest, you need to put in double the work that any other individual does

**Comment by steelwolf180** [comment]
Chuphay, I get what you mean and i appreciate your honesty. I just wanted everyone to be on the same page when this function is integrated to the application and to further enhance it when the time comes with new ideas or functions to move the project further. 

It is not just my project and it's ours every single one of us. The last thing I want you all is to waste your precious time on a function or idea that is not implemented into the project. At least we get to trash it out so before it is implemented. We could make the idea better and further make the project be in line with the project's purpose and the direction for the project to move forward as a whole.

Ps: I can't thank enough for what Jonathan Humphreys had done in the project by including the GUI, editing my codes and spelling errors.
